packages/react-core/src/components/Accordion/AccordionToggle.tsx

```diff
@@ -4,7 +4,8 @@ import styles from '@patternfly/react-styles/css/components/Accordion/accordion'
 import AngleRightIcon from '@patternfly/react-icons/dist/js/icons/angle-right-icon';
 import { AccordionContext } from './AccordionContext';
 
-export interface AccordionToggleProps extends Omit<React.HTMLProps<HTMLButtonElement>, 'type'> {
+export interface AccordionToggleProps
+  extends React.DetailedHTMLProps<React.ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement> {
```

Comment:
I have not seen this before but a quick google search says something about DetailedHTMLProps including ref and key as props? Should we be extending this in all our function components? Can you point me to why you made this update?

Reply:
I made the update because of the `type` prop. Without it passing 'submit' or 'reset' or 'button' gives an error (it's parsing as a string instead of one of the enum). I've seen it in a couple of components, but I'm unsure which extension we should use over.

Reply:
okay. I'll keep it in my back pocket as a possible remedy for other typescript errors/warnings in the future.